### PR TITLE
Issue5253

### PR DESCRIPTION
**Pull request by rahu**
this fixed 5253: edit personal note has done / cancel button on top. you can edit the note from optionsmenu and listing will not jump back to fist line after editing.

### CODE REVIEW

File: main/src/cgeo/geocaching/CacheDetailActivity.java [hunk 1]
```diff
@@ -2294,11 +2297,6 @@ protected Void doInBackground(final Void... params) {
                 DataStore.saveCache(cache, EnumSet.of(SaveFlag.DB));
                 return null;
             }
-
-            @Override
-            protected void onPostExecute(final Void v) {
-                notifyDataSetChanged();
-            }
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/CacheDetailActivity.java]: Could you explain why you removed this?

File: main/src/cgeo/geocaching/CacheMenuHandler.java [hunk 2]
```diff
@@ -88,6 +88,7 @@ public static void onPrepareOptionsMenu(final Menu menu, final Geocache cache) {
             return;
         }
         final boolean hasCoords = cache.getCoords() != null;
+		menu.findItem(R.id.menu_edit_fieldnote).setVisible(true);
```
Comment (SammysHP) [inline main/src/cgeo/geocaching/CacheMenuHandler.java]: Indentation.

File: main/src/cgeo/geocaching/CacheDetailActivity.java (hunk 1 above)
Comment (SammysHP) [inline main/src/cgeo/geocaching/CacheDetailActivity.java]: To "fix" the problem that the description page jumps back to the top after closing the dialog.

The reason for the jump is that the page is recreated with the new data. I assume that the content of the page won't be updated without `notifyDataSetChanged()`.

**Comment by rahu** [comment]
These causes ui to be redrawn and line one is back on top. You have to scroll down again.. And I haven't found a reason why not simply remove this.

File: main/src/cgeo/geocaching/CacheMenuHandler.java (hunk 2 above)
Comment (rahu) [inline main/src/cgeo/geocaching/CacheMenuHandler.java]: I hate eclipse...

File: main/src/cgeo/geocaching/CacheDetailActivity.java (hunk 1 above)
Comment (rahu) [inline main/src/cgeo/geocaching/CacheDetailActivity.java]: So is there a reason to leave it? In this case I'll find another solution.

Comment (SammysHP) [inline main/src/cgeo/geocaching/CacheDetailActivity.java]: Yes. The detail activity must know when the data was changed and when it has to update the UI.

Comment (rahu) [inline main/src/cgeo/geocaching/CacheDetailActivity.java]: But data (in this case personal note is updated without calling this).  The only difference I found after changing back is the "note has been saved" info which is not shown any longer  :( I'll fix this and come back...

**Comment by rahu** [comment]
ok guys - I updated the commit. Now scroll position is reset to the last position after updating UI.

**Comment by Lineflyer** [comment]
@rahu Did you see, that this has conflicts to merge? Maybe you should send a new PR based on current master.

**Pull request closed without merging**